### PR TITLE
All dialects: Allow `RESPECT NULLS`/`IGNORE NULLS` in window functions

### LINKED ISSUE #3375: redshift dialect unable to parse window function with `respect nulls`
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Parsing using `sqlfluff =0.13.1` `dialect=redshift` fails on a window function that includes `respect nulls` .

### Expected Behaviour

`sqlfluff parse test.sql` should parse the SQL successfully

### Observed Behaviour

`sqlfluff parse test.sql`

```
...
==== parsing violations ====
L:   2 | P:  17 |  PRS | Line 2, Position 17: Found unparsable section: 'respect nulls over (order by col2...'
```

### How to reproduce

* create a test.sql file 
```
select
    lead(col1, 1) respect nulls over (order by col2 asc)
from dual
```
* run sqlfluff parse test.sql

### Dialect

redshift, also fails with postgres

### Version

# pip3 list | grep -e dbt -e fluff
```
dbt-core                 1.1.0
dbt-extractor            0.4.1
dbt-postgres             1.1.0
dbt-redshift             1.1.0
sqlfluff                 0.13.1
sqlfluff-templater-dbt   0.13.1
```

# sqlfluff --version
```
sqlfluff, version 0.13.1
```

### Configuration

.sqlfluff
```
# For SQLFluff Rules reference, see:
# https://docs.sqlfluff.com/en/stable/rules.html#rules-reference
[sqlfluff]
templater = dbt
dialect = redshift
exclude_rules = L031

[sqlfluff:templater:jinja]
apply_dbt_builtins = True

# Some rules can be configured directly from the config common to other rules.
[sqlfluff:rules]
tab_space_size = 4
indent_unit = space
max_line_length = 150

# Individual rule config

###
[sqlfluff:rules:L011]
aliasing = implicit

[sqlfluff:rules:L010] # Keywords
capitalisation_policy = lower

[sqlfluff:rules:L014]  # Unquoted identifiers
extended_capitalisation_policy = lower
unquoted_identifiers_policy = all

[sqlfluff:rules:L030]  # Function names
extended_capitalisation_policy = lower

[sqlfluff:rules:L040]  # Null & Boolean Literals
capitalisation_policy = lower
```

.sqlfluffignore
```
target/
dbt_packages/
macros/
```

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by tunetheweb**
### Brief summary of the change made

Fixes #3375 

### Are there any other side effects of this change that we should be aware of?
Nope

### Pull Request checklist
- [x] Please confirm you have completed any of the necessary steps below.

- Included test cases to demonstrate any code changes, which may be one or more of the following:
  - `.yml` rule test cases in `test/fixtures/rules/std_rule_cases`.
  - `.sql`/`.yml` parser test cases in `test/fixtures/dialects` (note YML files can be auto generated with `tox -e generate-fixture-yml`).
  - Full autofix test cases in `test/fixtures/linter/autofix`.
  - Other.
- Added appropriate documentation for the change.
- Created GitHub issues for any relevant followup/future enhancements if appropriate.

### CODE REVIEW

File: src/sqlfluff/dialects/dialect_sparksql.py [hunk 1]
```diff
@@ -1631,27 +1631,6 @@ class LateralViewClauseSegment(BaseSegment):
     )
 
 
-class OverClauseSegment(ansi.OverClauseSegment):
```
Comment (barrywhart) [inline src/sqlfluff/dialects/dialect_sparksql.py:1634]: Why was this removed?

**Review by barrywhart — APPROVED** [review]

File: src/sqlfluff/dialects/dialect_sparksql.py (hunk 1 above)
Comment (tunetheweb) [reply src/sqlfluff/dialects/dialect_sparksql.py:1634]: Because the reason it was overridden was to add the thing we’re now adding in ANSI. So no need for override anymore.
